tests/test/object_safe.rs

```diff
@@ -0,0 +1,29 @@
+use super::*;
+
+/// The current implementation of RustIrDatabase::is_object_safe only checks for
+/// the #[object_safe] flag. This test will start failing once it performs a
+/// proper test because Foo should be object safe.
+#[test]
+fn by_default_not_object_safe() {
+    test! {
+        program {
+            trait Foo {}
+        }
+
+        goal { ObjectSafe(Foo) } yields { "No possible solution" }
+        goal { not { ObjectSafe(Foo) } } yields { "Unique" }
+    }
+}
+
+#[test]
+fn object_safe_flag() {
+    test! {
+        program {
+            #[object_safe]
+            trait Foo {}
+        }
+
+        goal { ObjectSafe(Foo) } yields { "Unique" }
+        goal { not { ObjectSafe(Foo) } } yields { "No possible solution" }
```

Comment:
We don't really need both a goal and a `not` goal. Negation and such is tested elsewhere and it so here it doesn't really *test* anything other than just adding another thing to have to think about when looking at the test.